### PR TITLE
More tweaks about thinpool devices

### PR DESCRIPTION
**Pull request by mulkieran**
Some minor tweaks to ThinPoolDev implementation and related.

These come up since CacheDev is more similar to ThinPoolDev than any other device implemented so far.

### CODE REVIEW

**Comment by mulkieran** [comment]
@agrover review?

File: src/types.rs [hunk 1]
```diff
@@ -28,10 +28,20 @@ use super::deviceinfo::{DM_NAME_LEN, DM_UUID_LEN};
...
+/// a DM meta device may store cache device or thinpool device metadata
+/// defined in drivers/md/persistent-data/dm-space-map-metadata.h as
+/// DM_SM_METADATA_BLOCK_SIZE.
 const META_BLOCK_SIZE: Sectors = Sectors(8);
 
+/// The maximum size of a metadata device.
+/// defined in drivers/md/persistent-data/dm-space-map-metadata.h as
+/// DM_SM_METADATA_MAX_BLOCKS.
+/// It seems that this is not the maximum size allowed, but only the maximum
+/// size that can be used.
```
Comment (agrover) [inline src/types.rs]: Can you clarify this a bit? "Allowed" vs "can be used".

**Comment by mulkieran** [comment]
@agrover review? I expect I would merge the final 8 commits into a single commit.

**Comment by mulkieran** [comment]
Whoops!

**Comment by mulkieran** [comment]
I expect I would merge the final 10 commits into a single commit.

**Comment by mulkieran** [comment]
Corresponding stratisd PR: https://github.com/stratis-storage/stratisd/pull/662.

**Comment by mulkieran** [comment]
Other than the squashing, I think this is ready to go. Please _do not merge_ until I do.

**Comment by agrover** [comment]
what do you think about adding a ThinPoolWorkingStatus::new() fn?

**Comment by mulkieran** [comment]
@agrover review please? Only difference in code is I added a note in the ```ThinPoolDev::status``` header comment explaining why the code didn't match the kernel docs (it's because the kernel docs are wrong, of course).

**Comment by agrover** [comment]
👍 , merging. You're going to post a patch to fix the kernel docs I hope?

**Pull request merged**